edweiss-app/components/courses/AddAssignment.tsx

```diff
@@ -0,0 +1,198 @@
+                            <TText testID={testIDs.dateText} ml={14} color={dateChanged ? 'text' : 'overlay0'}>{date.toDateString()}</TText>
+                        </TTouchableOpacity>
+                    </TView>
+
+                    <TView testID={testIDs.timeInput} backgroundColor='crust' borderColor='surface0' radius={14} flex={1} flexDirection='column' ml='sm'>
+                        <TText testID={testIDs.timeTitle} ml={16} mb={4} size={'sm'} pl={2} pt={'sm'} color='overlay2'>{t(`course:due_time_label`)}</TText>
+                        <TTouchableOpacity testID={testIDs.timeTouchableOpacity} onPress={() => setShowPickerTime(true)}
+                            pr={'sm'} pl={'md'} pb={'sm'}
+                            flexDirection='row' justifyContent='flex-start' alignItems='center'>
+                            <Icon testID={testIDs.timeIcon} name={timeIcon} size='md' color='overlay0' />
+                            <TText testID={testIDs.timeText} ml={10} color={timeChanged ? 'text' : 'overlay0'}>{date.toTimeString().split(':').slice(0, 2).join(':')}</TText>
+                        </TTouchableOpacity>
+                    </TView>
+                </TView>
+
+
+                {showPickerDate && (
+                    <DateTimePicker
+                        testID={testIDs.datePicker}
+                        value={date}
+                        mode='date'
+                        is24Hour={true}
+                        display="default"
+                        onChange={(event, selectedDate) => {
+                            if (event.type === "dismissed") {
+                                setShowPickerDate(false);
+                            } else {
+                                onChangeDate(event, selectedDate);
+                            }
+                        }}
+                    />
+                )}
+
+                {showPickerTime && (
+                    <DateTimePicker
+                        testID={testIDs.timePicker}
+                        value={date}
+                        mode='time'
+                        is24Hour={true}
+                        display="default"
+                        onChange={(event, selectedDate) => {
+                            if (event.type === "dismissed") {
+                                setShowPickerTime(false);
+                            } else {
+                                onChangeTime(event, selectedDate);
+                            }
+                        }}
+                    />
+                )}
+
+            </TScrollView >
+
+            <TTouchableOpacity
+                testID={testIDs.finishTouchableOpacity}
+                backgroundColor={(name === "" || !dateChanged || !timeChanged) ? 'text' : 'blue'}
+                disabled={name === "" || !dateChanged || !timeChanged}
+                onPress={() => { console.log("Data has to be updated in Firebase!"); onSubmit(); }}
```

Comment:
For quizzes, we'll have to redirect to the quiz creation page, with the name and due date passed down in the router. The submission logic of quizzes is handled in the creation page.

Reply:
Yep let's do that on the next sprint!